Stability fixes

vikulin opened the pull request.

mwarning commented:
wow, impressive.

A few questions:
- what features did you add?
- did your bugs did your fix?
- why are the commit messages starting with `1.`?
- what linter (how) did you run? I also like to apply that to the develoment branch.

mwarning commented on app/build.gradle:
```diff
@@ -3,15 +3,16 @@ apply plugin: 'com.android.application'
-        minSdkVersion 19
-        targetSdkVersion 26
-        versionCode 303
-        versionName "3.0.3"
```
We can set it to version 3.1.0.

vikulin commented:
> wow, impressive.
> 
> A few questions:
> 
>     * what features did you add?
Mostly this PR contains simplifications:
 - removed encryption since [RiV-mesh](https://github.com/RiV-chain/RiV-mesh) network supports it by default
 - removed MAC in interface list in Settings
 - upgraded and refactored WebRTC code
>     * did your bugs did your fix?
 - fixed unexpected interruption while calling
 >     * why are the commit messages starting with `1.`?
 - It's used for commit index
>     * what linter (how) did you run? I also like to apply that to the develoment branch.
 - Used AndroidStudio and tested on Android 5.0, Android 10 also showed good results

Generally this is preparation for re-brending and much more deep changes in GUI. If you prefer to keep the same name (meshenger) let discuss it in [telegram channel](https://t.me/rivchain).

mwarning commented:
I see, but this cannot be merged here when removing encryption and mac support. It might be a good idea to make all this optional. But you can do whatever you want. :-)

vikulin commented:
You might take several ideas and merge partially/manually. I will keep the
project as a fork respecting your afforts as a creator.

чт, 25 нояб. 2021 г. в 22:20, Moritz Warning ***@***.***>:

> I see, but this cannot be merged here when removing encryption and mac
> support. It might be a good idea to make all this optional. But you can do
> whatever you want. :-)
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/meshenger-app/meshenger-android/pull/55#issuecomment-979453728>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAFVRRXI37U2QCQOYCQ5NJDUN2LBBANCNFSM5IY33LIQ>
> .
>

mwarning commented:
I do not mind if you want to remove the fork relationship. I will see what bugfixes and features I can merge and say thank you.

mwarning commented on app/src/main/java/d/d/meshenger/Utils.java:
```diff
@@ -289,8 +294,14 @@ public static boolean isIP(String address) {
                     if(fullAddress.indexOf('%')>0){
                         String subaddress = fullAddress.substring(0, fullAddress.indexOf('%'));
+                        if(isIPv6(subaddress)){
+                            subaddress="["+subaddress+"]";
```
Is this a problem somewhere?

vikulin commented:
Yes, interface address may contains % symbol

чт, 25 нояб. 2021 г. в 23:36, Moritz Warning ***@***.***>:

> ***@***.**** commented on this pull request.
> ------------------------------
>
> In app/src/main/java/d/d/meshenger/Utils.java
> <https://github.com/meshenger-app/meshenger-android/pull/55#discussion_r757133763>
> :
>
> > @@ -289,8 +294,14 @@ public static boolean isIP(String address) {
>                      addressList.add(new AddressEntry(addr.getHostAddress(), nif.getName(), addr.isMulticastAddress()));
>                      if(fullAddress.indexOf('%')>0){
>                          String subaddress = fullAddress.substring(0, fullAddress.indexOf('%'));
> +                        if(isIPv6(subaddress)){
> +                            subaddress="["+subaddress+"]";
>
> Is this a problem somewhere?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/meshenger-app/meshenger-android/pull/55#pullrequestreview-816304290>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAFVRRUGJ25RL4D5G7HBKPDUN2T37ANCNFSM5IY33LIQ>
> .
>

vikulin commented:
> I see, but this cannot be merged here when removing encryption and mac support. It might be a good idea to make all this optional. But you can do whatever you want. :-)

WebRTC from Google should support encryption. Isn't it?

mwarning commented:
WebRTC itself is encrypted, but the initial key exchange is not.

vikulin commented:
Are you referring to a MITM attack?

пт, 26 нояб. 2021 г. в 09:55, Moritz Warning ***@***.***>:

> WebRTC itself is encrypted, but the initial key exchange is not.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/meshenger-app/meshenger-android/pull/55#issuecomment-979762447>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAFVRRWCPAPGVQ2NRWG2DIDUN44PNANCNFSM5IY33LIQ>
> .
>

mwarning commented:
Yes, for example MIM.

opusforlife2 commented:
@mwarning Will you be releasing a new version with code picked from here, then?

mwarning commented:
@opusforlife2 I will try to integrate some things into the development branch which hopefully turn into a new release. But the problem is available time..

vikulin commented:
> @opusforlife2 I will try to integrate some things into the development branch which hopefully turn into a new release. But the problem is available time..

Consider kotlin migration for all code which I have already done:

https://github.com/meshenger-app/meshenger-android/pull/55/files

opusforlife2 commented on app/build.gradle:
```diff
@@ -23,22 +28,27 @@ android {
-    implementation 'com.github.joshjdevl.libsodiumjni:libsodium-jni-aar:2.0.2'
-    implementation 'net.java.dev.jna:jna:4.5.2@aar'
-    implementation 'com.github.isabsent:filepicker:1.1.01'
+    //back to 3.3.0 instead of 3.4.0 sue to https://github.com/zxing/zxing/issues/1170
```
```suggestion
    //back to 3.3.0 instead of 3.4.0 due to https://github.com/zxing/zxing/issues/1170
```

mwarning commented:
> > @opusforlife2 I will try to integrate some things into the development branch which hopefully turn into a new release. But the problem is available time..
> 
> Consider kotlin migration for all code which I have already done:
> 
> https://github.com/meshenger-app/meshenger-android/pull/55/files

I always wanted to migrate to Kotlin. :-)

vikulin commented:
Done release [v0.4.0.7](https://github.com/RiV-chain/CupLink/releases/tag/v0.4.0.7):

1.Tested Kotlin version
2.Redesigned Choose Address form
3.Bufixes

mwarning commented:
Hi @vikulin, I started to integrate some of your fixes into the repository.
Let's close this MR for now, since it won't be merged directly.
